Contributions/CheckList_for_Safety_Claims_on_a_Generic_Linux_System.md

```diff
@@ -102,6 +102,24 @@ And because of the variance in the output of the very same toolchain, based on t
 
 It is therefore necessary, as preliminary step, to verify that the toolchain is qualified accordingly to safety requirements, and used accordingly to its qualification. Lacking this, then it becomes necessary to identify an adequate mitigation strategy.
 
+
+## **Other Execution Contexts**
+
+This section is particularly tied to the hardware configuration that one might have chosen (see reference architecture document and related interference).
+On modern processors it is common to have additional hardware contexts, which expose low level functionality that is not available to the operating system.
+From a very high level perspective, these modes bring two capability:
+  * Hypervisor Mode: just like the operating system can run multiple applicaitons in parallel, gating their access to certain features, isolating them from each other and abstracting the hardware, so the hypervisor can execute in parallel multiple operating systems, controlling their access to resources and abstracting the hardware. And, just like an application can do little to protect itself fro mthe operating system, so an operating system can do very little to protect itself from the hypervisor.
+  * Secure Mode: sometimes it is necessary to provide strong guarrantees that certain data (e.g cryptographyc private keys or oter secrets) is not accessible to the rest of the system (which might be compromised); this ensures also that extra efforts can be put in developing the code which *does* have access. The secure mode ensures that a system designer can always rely on certain ground truths.
+
+The way these additional contexts are specified and implemented also grants them almost unfettered access to the kernel memory. The kernel might not even be aware of running on top of an hypervisor, or that a secure mode is mediating its access to certain memory.
+Needless to say, these priviliges provide also an unbound ability to interfere.
+
+**Can the kernel do anything to protect itself against this interference?** Possibly, but it would ome very close to sprinkling checksums and redundancy all over. This would mean diverging significantly from a vanilla kernel *and* also introducing a noticeable overhead.
+
+**What else can be done?** It depends. One might attempt to rely on significant redundancy and then deploy some form of safe monitor, responsible of comparing the behavior of the paired systems. Alternatively, one could qualify these additional contexts which have the ability to interfere. Considering that usually these contexts are already subject to very rigorous process, because they can do so much damage, if buggy, it might be easier to qualify them, than it would be to qualify the Linux kernel. Furthermore, they tend to be more specialised pieces of software, implementing less functionality, with less code, than a full blown operating system.
```

Comment:
```suggestion
**What else can be done?** It depends. One might attempt to rely on significant redundancy and then deploy some form of safe monitor, responsible for comparing the behaviour of the paired systems. Alternatively, one could qualify these additional contexts that have the ability to interfere. Considering that usually these contexts are already subject to a very rigorous process, because they can do so much damage if buggy, it might be easier to qualify them than to qualify the Linux kernel. Furthermore, they tend to be more specialised pieces of software, implementing less functionality, with less code, than a full-blown operating system.
```